apps/web/src/pages/settings/user-profile-page/UserProfileUpdatePasswordForm.tsx

```diff
@@ -0,0 +1,145 @@
+import { showNotification } from '@mantine/notifications';
+import { Button, IconOutlineLockPerson, PasswordInput } from '@novu/design-system';
+import { checkIsResponseError, IResponseError } from '@novu/shared';
+import { api } from '@novu/shared-web';
+import * as Sentry from '@sentry/react';
+import { useMutation } from '@tanstack/react-query';
+import { useForm } from 'react-hook-form';
+import { css } from '../../../styled-system/css';
+import { Stack } from '../../../styled-system/jsx';
+import { PasswordRequirementPopover } from '../../auth/components/PasswordRequirementPopover';
+import { SHARED_PASSWORD_INPUT_REGISTER_OPTIONS } from './UserProfilePasswordSidebar.shared';
+
+type UserProfileUpdatePasswordFormProps = {
+  onSuccess?: () => void;
+};
+
+interface IPasswordUpdateData {
+  currentPassword: string;
+  newPassword: string;
+  confirmPassword: string;
+}
+
+export const UserProfileUpdatePasswordForm: React.FC<UserProfileUpdatePasswordFormProps> = ({ onSuccess }) => {
+  const { isLoading, mutateAsync, error, isError } = useMutation<
+    IPasswordUpdateData,
+    IResponseError,
+    IPasswordUpdateData
+  >((data) => api.post(`/v1/auth/update-password`, data));
+
+  const {
+    register,
+    handleSubmit,
+    control,
+    formState: { errors, isValid },
+    setError,
+    reset,
+  } = useForm({
+    defaultValues: {
+      currentPassword: '',
+      newPassword: '',
+      confirmPassword: '',
+    },
+  });
+
+  const onSubmitPasswords = async (data: IPasswordUpdateData) => {
+    if (data.newPassword !== data.confirmPassword) {
+      setError('confirmPassword', { message: 'Passwords do not match' });
+
+      return;
+    }
+
+    try {
+      await mutateAsync(data);
+
+      showNotification({
+        message: 'Password was set successfully',
+        color: 'green',
+      });
```

Comment:
<!-- TODO's are small, trivial, but necessary changes. Distinguishing todo comments from issues: or suggestions: helps direct the reader's attention to comments requiring more involvement.  -->
**todo:** Please use the `errorMessage` and `successMessage` notification helpers from our design-system‏